CommonUI/Sources/CommonUI/MovieUI/MovieDetails/MovieDetailsView.swift

```diff
@@ -2,13 +2,13 @@ import SwiftUI
 import ApiService
 import UserData
 
-public struct MovieDetails: View {
-    let movie: Movie
-    @State private var isFavorite: Bool
+public struct MovieDetailsView: View {
+    @ObservedObject private var viewModel: MovieDetailsViewModel
+    private let movie: Movie
     
-    public init(movie: Movie, isFavorite: Bool) {
+    public init(movie: Movie) {
```

Comment:
Suggest init() using dependency injection for better testing.

    public init(viewModel: MovieDetailsViewModel) {
        self.viewModel = viewModel
    }
    
You can get rid of `private let movie: Movie`

Then change the code in two places with function createMovieDetailsView :

    func createMovieDetailsView(movie: Movie) -> MovieDetailsView {
        let viewModel = MovieDetailsViewModel(movie: movie)
        return MovieDetailsView(viewModel: viewModel)
    }
    
    Make init public in `MovieDetailsViewModel`
        public init(movie: Movie) {
        self.movie = movie
        self.fetchFavorite()
    }

Reply:
As our discussion, we can init the viewModel inside the view since the movie object can inject here.